src/Git/CommitFileViaConsole.php

```diff
@@ -26,4 +29,30 @@ public function __invoke(
             $repositoryDirectory
         ))->mustRun();
     }
+
+    /**
+     * @param non-empty-string $repositoryDirectory
+     * @param non-empty-string $filename
+     */
+    private function assertBranch(
+        BranchName $expectedBranch,
+        string $repositoryDirectory,
+        string $filename
+    ): void {
+        $process = new Process(['git', 'branch', '--show-current'], $repositoryDirectory);
+        $process->run();
```

Comment:
You want to call `$process->mustRun()` here: that already throws meaningful exceptions with STDOUT/STDERR as part of them (no need for the assertion either)